scipy/signal/tests/test_signaltools.py

```diff
@@ -1997,34 +2197,62 @@ def _setup_rank3(self, dt):
 
         return a, b, y_r
 
-    def test_rank3_valid(self, dt):
-        a, b, y_r = self._setup_rank3(dt)
+    @skip_xp_backends(np_only=True, reasons=["order='F'"])
+    def test_rank3_valid(self, dt, xp):
+        dt = getattr(xp, dt) if isinstance(dt, str) else dt
+        a, b, y_r = self._setup_rank3(dt, xp)
         y = correlate(a, b, "valid")
-        assert_array_almost_equal(y, y_r[1:2, 2:4, 3:5])
-        assert_equal(y.dtype, dt)
+        _assert_almost_equal = self._get_assertion(dt)
+        _assert_almost_equal(y, y_r[1:2, 2:4, 3:5])
+        assert y.dtype == dt
 
         # See gh-5897
         y = correlate(b, a, "valid")
-        assert_array_almost_equal(y, y_r[1:2, 2:4, 3:5][::-1, ::-1, ::-1])
-        assert_equal(y.dtype, dt)
-
-    def test_rank3_same(self, dt):
-        a, b, y_r = self._setup_rank3(dt)
+        _assert_almost_equal = self._get_assertion(dt)
+        _assert_almost_equal(y, y_r[1:2, 2:4, 3:5][::-1, ::-1, ::-1])
+        assert y.dtype == dt
+
+    @skip_xp_backends(np_only=True, reasons=["order='F'"])
+    def test_rank3_same(self, dt, xp):
+        dt = getattr(xp, dt) if isinstance(dt, str) else dt
+        a, b, y_r = self._setup_rank3(dt, xp)
         y = correlate(a, b, "same")
-        assert_array_almost_equal(y, y_r[0:-1, 1:-1, 1:-2])
-        assert_equal(y.dtype, dt)
-
-    def test_rank3_all(self, dt):
-        a, b, y_r = self._setup_rank3(dt)
+        _assert_almost_equal = self._get_assertion(dt)
+        _assert_almost_equal(y, y_r[0:-1, 1:-1, 1:-2])
+        assert y.dtype == dt
+
+    @skip_xp_backends(np_only=True, reasons=["order='F'"])
+    def test_rank3_all(self, dt, xp):
+        dt = getattr(xp, dt) if isinstance(dt, str) else dt
+        a, b, y_r = self._setup_rank3(dt, xp)
         y = correlate(a, b)
-        assert_array_almost_equal(y, y_r)
-        assert_equal(y.dtype, dt)
+        _assert_almost_equal = self._get_assertion(dt)
+        _assert_almost_equal(y, y_r)
+        assert y.dtype == dt
+
+
+@pytest.mark.parametrize('dt', ["uint8", "int8", "uint16", "int16",
+                                "uint32", "int32", # np_ulong, np_long,   # FIXME: np 1.x & 2.x / backends?
+                                "uint64", "int64", # FIXME: is this correct? Was np.ulonglong, np.ulonglong,
+                                "float32", "float64",
+                                # np.longdouble, # FIXME: what to do w/ longdouble?
```

Comment:
For now can keep (either with skips in the test, or separate out the longdouble test). It may make sense to deprecate and remove though, since our `longdouble` support is very patchy and generally not useful (internally things are regular `double` almost always anyways).